pkg/metrics/metrics.go

```diff
@@ -0,0 +1,52 @@
+package metrics
+
+import (
+	"fmt"
+	logger "github.com/intel/cri-resource-manager/pkg/log"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	builtInCollectors map[string]InitCollector
+	log               = logger.NewLogger("metrics")
+)
+
+func init() {
+	builtInCollectors = make(map[string]InitCollector)
+}
+
+type InitCollector func() (prometheus.Collector, error)
+
+func RegisterCollector(name string, init InitCollector) error {
+	log.Info("registering collector %s...", name)
+
+	if _, fn := builtInCollectors[name]; fn {
```

Comment:
`fn` is often meant to represent `function`. Perhaps `found` would be clearer here.

Reply:
will change